### PR TITLE
Add support for non-matrix projects to be aggregated in and depended on by matrix projects

### PR DESCRIPTION
**Pull request by vilunov**

### CODE REVIEW

**Review by eed3si9n — COMMENTED** [review]
Thanks for the contribution!

File: src/main/scala/sbt/internal/ProjectMatrix.scala [hunk 1]
```diff
@@ -249,7 +259,9 @@ object ProjectMatrix {
         IO.createDirectory(dotSbtMatrix)
         val p = Project(childId, dotSbtMatrix / childId)
           .dependsOn(deps: _*)
+          .dependsOn(nonMatrixDependencies: _*)
```
Comment (eed3si9n) [inline src/main/scala/sbt/internal/ProjectMatrix.scala]: For the clarity of the code, do you mind appending `nonMatrixDependencies` to the above `deps` at line 252 instead?

File: src/main/scala/sbt/internal/ProjectMatrix.scala [hunk 2]
```diff
@@ -249,7 +259,9 @@ object ProjectMatrix {
...
           .aggregate(aggs: _*)
+          .aggregate(nonMatrixAggregate: _*)
```
Comment (eed3si9n) [inline src/main/scala/sbt/internal/ProjectMatrix.scala]: Same here but with `aggs`.

**Review by eed3si9n — APPROVED** [review]

**Comment by vilunov** [comment]
Thanks!

**Pull request merged**